### PR TITLE
Added usage warnings

### PR DESCRIPTION
**Pull request by mauhiz**

### CODE REVIEW

**Comment by lloydmeta** [comment]
This looks good; I think it would be nicer if you just went with type classes, given that you're already 99.9% of the way there.

**Comment by mauhiz** [comment]
> type classes

Not sure what you mean, could you please show me a sample?

**Comment by lloydmeta** [comment]
Sorry, I mean that the [ConfigChecker[T]](https://github.com/m3dev/octoparts/pull/138/files#diff-2a61808e73c3eb5e5e219ec575edfb45R12) is already a type class :)

A ConfigChecker  instance could then be typed according to the kind of object it checks, for example

```scala
implicit object QueryChecker extends ConfigChecker[PathParam] {
  // check for stuff here
}
```

The ConfigChecker singleton can then expose something like 

```scala
def check[T <: ConfigChecker](obj: T)(implicit lang: Lang): String = implicitly[ConfigChecker[T]]. warnings(obj)
```

Then you can traverse a `HttpPartConfig` and check the individual kinds of objects easily without having to pass the whole HttpPartConfig around.

Thats how typeclasses can be used here :) Not a big deal, I would happily merge this as is; just thought it might be something fun :)

For more info on typeclasses, [check out this great blog](http://danielwestheide.com/blog/2013/02/06/the-neophytes-guide-to-scala-part-12-type-classes.html)

**Comment by mauhiz** [comment]
OK, that implies is that we we wouldn't be able to have several distinct checkers with the same type parameter? In which case I think we should keep it as is, for readability

**Comment by lloydmeta** [comment]
It really depends on how you think about it; the other way is to say that there are not enough types to describe the problem space and increase the types we have (e.g. PathParams, QueryParams). That would keep or increase readability by using types for self-documenting code.

**Comment by mauhiz** [comment]
I see. Adding that many new types would be a prowess, but maybe not needed yet :)

File: app/controllers/support/ConfigChecker.scala [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import java.util.UUID
+
+import com.m3.octoparts.aggregator.handler.HttpPartRequestHandler
+import com.m3.octoparts.model.config.{ HttpPartConfig, ParamType, PartParam }
+import com.netaporter.uri.Parameters.Param
+import com.netaporter.uri.dsl._
+import play.api.{ Mode, Play }
+import play.api.i18n.{ Lang, Messages }
+
+trait ConfigChecker[T] {
```
Comment (lloydmeta) [inline app/controllers/support/ConfigChecker.scala]: It looks like the [T] parameter is pointless here?

Comment (mauhiz) [inline app/controllers/support/ConfigChecker.scala]: For now the only implementations have T = HttpPartConfig, but in the near future I am sure there will be more

Comment (lloydmeta) [inline app/controllers/support/ConfigChecker.scala]: Not needed yet :) I would prefer it not to have the type param unless it actually get's used. A type alias inside the trait would be more appropriate here I think.

Comment (mauhiz) [inline app/controllers/support/ConfigChecker.scala]: done

**Pull request merged**